CONTRIBUTING.md

```diff
@@ -0,0 +1,105 @@
+# Contributing to Yarn
+
+## What's in a Name?
+
+The fundamental goal of mappings is to make modding Minecraft easier to understand, for beginners and experts alike. We want
+to create a cohesive, simple naming scheme that avoids confusion or being too verbose. Having a lot of things named in
+different schemes would be a nightmare for usability, so we want to avoid that. This includes naming schemes not just within
+Yarn but within the greater Minecraft community. Despite that goal, there are some name sources that are inherently more
+"reliable" than others; if an official term is hinted at or can be guessed, you should generally use that - sticking closer to 
+official terminology helps everyone.
+
+## Guidelines
+
+To keep with the goal of consistency within the mappings and the greater Minecraft community, here are a set of guidelines to
+follow when naming unmapped classes, fields, methods, method arguments and other aspects of the code.
+
+### 1. When known, stick to official terminology.
+
+This can be a contentious decision, due to many of Notch's original names for things (that he himself has stated were
+terrible choices for names) sticking around. While it could be nice to have names like `Gui` or the like, we have decided to
+use `Screen`, since that's the name for guis that the client-side registry uses. Official terms can be found in many places,
```

Comment:
`Screen` is a bad example. In this case the Screen name is much better than Gui as it actually is a screen (it fills the whole screeb and only one can be open at once), and non-screen GUIs are called huds or something else.

Reply:
Oh, that is a good point for differentiation between the two. It was just the only example I could think of where we changed a yarn name after learning the official name.

Reply:
Although I agree with this rule, I don't think it's what we're currently doing. For example, it's known from an exception message that Mojang doesn't use an Entity suffix for mobs, and several other people think dropping the suffix would be good, yet we still keep it.

Reply:
@Boundarybreaker A better example would be "World -> Level". But as I said in my last comment, it seems like this rule about using known Mojang terms isn't being applied, so adding it in the guidelines will just confuse people.

Reply:
not the place to have these discussions, really.

Reply:
How is it not? I'm saying this rule is inaccurate and should be removed.

Reply:
>For example, it's known from an exception message that Mojang doesn't use an Entity suffix for mobs, and several other people think dropping the suffix would be good, yet we still keep it.

Firstly, I haven't said no; secondly, I have not seen that exception message.

>A better example would be "World -> Level".

That, likewise, is still open - for this exact reason.

>How is it not? I'm saying this rule is inaccurate and should be removed.

Edited to reflect reality, but not removed - notice things like enum names, field names, many method names and class names do actually follow it.

Reply:
> I have not seen that exception message.

In the original draft for this, I said that we'll only move to known official names when we have a source for knowing the official names, and that people proposing a change like that should provide sources. I think that part should be put back in.